osprey_flask_app/utils.py

```diff
@@ -21,105 +22,152 @@ def get_filename_from_path(path):
+
+    get_input_files(arg_dict)
+    create_pour_points(arg_dict)
     return arg_dict
 
 
 def inputs_are_valid(arg_dict):
-    """Check that start/stop dates have a proper format and that filepaths exist either on
-    THREDDS or in local storage.
+    """Check that start/stop dates have a proper format, all pour points have (lon, lat, name), and filepaths exist on THREDDS.
     Parameters
-        1. arg_dict (dict): arguments supplied to osprey with corresponding values. It should contain
+        1. arg_dict (dict): arguments supplied to osprey with corresponding values. This function checks
         the following keys.
 
-            1. case_id (str): Case ID for the RVIC process
-            2. grid_id (str): Routing domain grid shortname
-            3. run_startdate (str): Run start date (yyyy-mm-dd-hh). Only used for startup and drystart runs.
-            4. stop_date (str): Run stop date.
-            5. pour_points (path): Comma-separated file of outlets to route to [lons, lats]
-            6. uh_box (path): Defines the unit hydrograph to route flow to the edge of each grid cell.
-            7. routing (path): Routing inputs netCDF.
-            8. domain (path): CESM compliant domain file.
-            9. input_forcings (path): Land data netCDF forcings.
-            10. loglevel (str): Logging level (one of 'CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG', 'NOTSET').
-                Default is 'INFO'.
-            11. version (int): Return RVIC version string (1) or not (0). Default is 1.
-            12. np (int): Number of processors used to run job. Default is 1.
-            13. params_config_file (path): Path to input configuration file Parameters process.
-            14. params_config_dict (str): Dictionary containing input configuration for Parameters process
-                (mutually exclusive with params_config_file).
-            15. convolve_config_file (path): Path to input configuration file Convolution process.
-            16. convolve_config_dict (str): Dictionary containing input configuration for Convolution process
-                (mutually exclusive with convolve_config_file).
+            1. run_startdate (str): Run start date. Only used for startup and drystart runs.
+            2. stop_date (str): Run stop date.
+            3. pour points (str): Outlets to route to [lons, lats]
+            4. uh_box (path): Defines the unit hydrograph to route flow to the edge of each grid cell.
+            5. routing (path): Routing inputs netCDF.
+            6. domain (path): CESM compliant domain file.
+            7. input_forcings (path): Land data netCDF forcings.
     """
 
     # Check start/stop dates
     parse(arg_dict["run_startdate"])
     parse(arg_dict["stop_date"])
 
+    # Check pour points
+    pour_points = arg_dict["pour_points"].split("\n")
+    pour_points = pour_points[1:]  # Do not check header
+    for point in pour_points:
+        if "None" in point:
```

Comment:
This seems like a very specific and yet insufficient way to check for validity. Maybe it'd be better to try to convert each set of pour point to a pair of `float`s, and if that fails then they must be invalid?